File: Angular/nearest small.java
 import java.util.*;

public class NearestSmallestValues {
    public static int[] nearestSmallestValues(int[] arr) {
        Stack<Integer> stack = new Stack<>();
        int[] result = new int[arr.length];

        for (int i = 0; i < arr.length; i++) {
            while (!stack.empty() && stack.peek() >= arr[i]) {
                stack.pop();
            }

            if (stack.empty()) {
                result[i] = -1;
            } else {
                result[i] = stack.peek();
            }

            stack.push(arr[i]);
        }
        return result;
    }
    public static void main(String[] args) {
        int[] arr = {5, 3, 1, 9, 7, 3, 4, 1};
        int[] result = nearestSmallestValues(arr);
        System.out.println(Arrays.toString(result)); // Output: [-1, -1, -1, 1, 1, 1, 3, 1]
    }
}

File: Angular/nearest small.js
function nearestSmallestValues(arr) {
    const stack = [];
    const result = [];
  
    for (let i = 0; i < arr.length; i++) {
      while (stack.length > 0 && stack[stack.length - 1] >= arr[i]) {
        stack.pop();
      }
  
      if (stack.length === 0) {
        result.push(-1);
      } else {
        result.push(stack[stack.length - 1]);
      }
  
      stack.push(arr[i]);
    }
  
    return result;
  }
  
  const arr = [5, 3, 1, 9, 7, 3, 4, 1];
  const result = nearestSmallestValues(arr);
  console.log(result); // Output: [-1, -1, -1, 1, 1, 1, 3, 1]